### ISSUE TITLE
[Feature]: Enable shift + click in journal

### ISSUE THREAD

**Issue by Pro-Panda** [sugarlabs/sugar#786]
**Proposed Feature:**
 1. Open the journal
 2. Click on a checkbox(`CellRendererToggle`) next to an activity (say, 2nd in line from the top)
 3. Hold `shift` and click on another checkbox (say, 5th in line from the top)
 
**Observed output:**
 1. 2nd gets selected
 2. 5th is not selected (Shift + click is not detected)

**Proposed output:**
 2nd, 3rd, 4th and 5th activities should get selected(checked)

**Reason for proposal:**
This could be helpful when a user wishes to remove a continuous list of activities(say, 10) out of a total of (say. 30). This coincides with the general way of selection using `shift` key

**Finding:**
 1. `Shiift` + `click` is not detected by `toggled` signal. It has no other signals available.
 2.  A `SHIFT Mask` check can't be done because of `.
 3. `TreeView Column` also didn't give a lead.

**Seeking guidance:**
 1. Whether this idea should be implemented ?
 2. If yes, what could be a possible area to look into
 3. This should be doable for favorite buttons(since they detect `shift`+`click`), but this again depends on 1. and the fact that is it achievable for checkboxes(to ensure consistency) without a cost

Kindly suggest

**Comment by quozl** [comment]
Yes, `shift+click` in a `Gtk.ListBox` would normally extend the selection through a range, but I don't know that it is common for lists that are tables with toggle buttons.  In lists, `click` deselects all others, and `ctrl+click` toggles an individual entry.  To be consistent we would need to bring these as well.  It would be a significant change, and would break the `click` which resumes the entry.

Workflow for deleting journal objects by name or time is to use the selection criteria first, then select one, then press the select all button.  Does this work for you?

We have a number of other problems with the journal if you're interested; I've bought a couple over into https://github.com/sugarlabs/sugar/issues/788.

**Comment by Pro-Panda** [comment]
An example of the proposed feature in lists would be list of mails in 'gmail', where a `click` toggles an individual entry, `ctrl+click` does the same thing, and `shift+click` does the proposed.

The selection criteria does the work well for me.

I agree that this might not be something we need, considering that we have more pressing issues already.

Thanks for pointing out the other issues. I will try to develop a fix for them

**Comment by quozl** [comment]
So the proposal is adding `shift+click` to extend selection through a range, and `ctrl+click` to toggle an entry.  Post to sugar-devel@ with the [DESIGN] subject tag to engage with others.  Confine your post to the user interface or user experience dimension of the problem; don't get into implementation details.  Ignore any unrelated discussion on the thread that ensues.

For context, this was not a problem before [multiple selection feature](https://wiki.sugarlabs.org/go/Features/Multi_selection) was added, so it could be seen as an improvement to that feature.

**Comment by quozl** [comment]
Let's give it a week in the hope of meaningful on-topic replies.  If we don't get anything by 7th March, let's go ahead.

**Comment by Pro-Panda** [comment]
Sure. 
May we start after 11th ? (I will have exams until then)

**Comment by quozl** [comment]
No rush.  I'm fine with adding this feature before about October.  :grin:

**Comment by saiatl2354** [comment]
Has this feature been implemented yet?

**Comment by chimosky** [comment]
> Has this feature been implemented yet?

Seeing as the issue is still open, I'd say no.
